File: solutions/csharp/perfect-numbers/PerfectNumbers.cs
public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number <= 0)
            throw new ArgumentOutOfRangeException(
                nameof(number),
                number,
                "Number must be greater than zero.");
        
        int aliquot = AliquotSum(number);

        if (number < aliquot)
            return Classification.Abundant;
        if (number > aliquot)
            return Classification.Deficient;
        return Classification.Perfect;
    }

    private static int AliquotSum(int number) =>
        Enumerable
            .Range(1, number / 2)
            .Where(n => number % n == 0)
            .Sum();
}

File: solutions/python/perfect-numbers/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """

    if number <= 0:
        raise ValueError('Classification is only possible for positive integers.')

    aliquot_sum = aliquot_sum_of(number)
    if number < aliquot_sum:
        return 'abundant'
    if number > aliquot_sum:
        return 'deficient'
    return 'perfect'


def aliquot_sum_of(number):   
    return sum(x for x in range(1, number) if not number % x)
